### PR TITLE
Add Alcatel-Lucent Enterprise driver (aos6 & aos8)

### PR DESCRIPTION
**Pull request by jefvantongerloo**
# Description

Adds driver for Alcatel-Lucent Enterprise aos6 and aos8 devices.

# Tested against:
- aos6: 6.7.2.89.R06
- aos8: 8.6.289.R01

## Type of change

- [ x ] New feature (non-breaking change which adds functionality)
- [ x ] This change requires a documentation update

### CODE REVIEW

File: scrapli_community/alcatel/aos/alcatel_aos.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+SCRAPLI_PLATFORM = {
+    "driver_type": "generic",
+    "defaults": {
+        # Prompt can be anything, but best practice is to end with > or #
+        "comms_prompt_pattern": r"^[a-zA-Z0-9.\-_@\/:]{1,63}[>|#]\s*",
+        "sync_on_open": None,
+        "async_on_open": None,
+        "sync_on_close": default_sync_on_close,
+        "async_on_close": default_async_on_close,
+        "transport": "ssh2",
```
Comment (carlmontanari) [inline scrapli_community/alcatel/aos/alcatel_aos.py:15]: any reason to explicitly set ssh2 as transport? the whole point of community is so folks can have stuff pre canned to what they need/want so I'm totally good with it, just curious!

File: scrapli_community/alcatel/aos/alcatel_aos.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+from scrapli_community.alcatel.aos._async import default_async_on_close
+from scrapli_community.alcatel.aos.sync import default_sync_on_close
+
+SCRAPLI_PLATFORM = {
+    "driver_type": "generic",
+    "defaults": {
+        # Prompt can be anything, but best practice is to end with > or #
+        "comms_prompt_pattern": r"^[a-zA-Z0-9.\-_@\/:]{1,63}[>|#]\s*",
+        "sync_on_open": None,
```
Comment (carlmontanari) [inline scrapli_community/alcatel/aos/alcatel_aos.py:11]: I think from our slack convo I remember seeing something similar to `term len 0` is that not generally needed?

**Comment by carlmontanari** [comment]
I dropped two little question/comments in there. If you could also add this to the table in the README that would be awesome! Thanks a bunch for the work on this!

File: scrapli_community/alcatel/aos/alcatel_aos.py (hunk 1 above)
Comment (jefvantongerloo) [reply scrapli_community/alcatel/aos/alcatel_aos.py:15]: Paramiko also works, but I though ssh2 to be the most performant option:
https://parallel-ssh.org/post/parallel-ssh-libssh2/

File: scrapli_community/alcatel/aos/alcatel_aos.py (hunk 2 above)
Comment (jefvantongerloo) [reply scrapli_community/alcatel/aos/alcatel_aos.py:11]: I tested both and term len 0 is not required, it was the missing "^" that caused the issue :)

File: scrapli_community/alcatel/aos/alcatel_aos.py (hunk 1 above)
Comment (carlmontanari) [reply scrapli_community/alcatel/aos/alcatel_aos.py:15]: cool. yeah any of the transports *should* work, but ssh2 is for sure fast! 

Also, heads up: there is no wheel for Darwin for 3.10 yet so just keep that in mind! Lastly -- ssh2 is another requirement for users who want to use this platform w/ defaults (whereas there are literally *no* requirements for `system` transport) -- I'm good with it if you are, just wanted to call it out!

**Comment by carlmontanari** [comment]
Ah dude I'm sorry -- I just messed you up a bit. I haven't merged develop to master for like an eternity and forgot that the README stuff moved into the docs and the README is now just super basic simple with links to the GitHub pages docs and such. If ya can revert that readme change I will handle updating the docs and stuff since I sent you astray :) Sorry about that man!

**Comment by carlmontanari** [comment]
superseded by #59

**Pull request closed without merging**